crates/aranya-policy-lang/src/lang/parse.rs

```diff
@@ -1551,20 +1556,35 @@ pub fn parse_ffi_decl(data: &str) -> Result<ast::FunctionDecl, ParseError> {
     Ok(fn_decl)
 }
 
-/// Parse a series of Struct definitions for the FFI
-pub fn parse_ffi_structs(data: &str) -> Result<Vec<AstNode<ast::StructDefinition>>, ParseError> {
-    let def = PolicyParser::parse(Rule::ffi_struct_def, data)?;
+/// A series of Struct or Enum definitions for the FFI
+pub type FfiTypes = (
+    Vec<AstNode<ast::StructDefinition>>,
+    Vec<AstNode<ast::EnumDefinition>>,
+);
+
+/// Parse a series of Struct or Enum definitions for the FFI
+pub fn parse_ffi_structs_enums(data: &str) -> Result<FfiTypes, ParseError> {
+    let def = PolicyParser::parse(Rule::ffi_struct_or_enum_def, data)?;
```

Comment:
nit: naming and return struct

```suggestion
pub struct FfiTypes {
    pub structs: Vec<AstNode<ast::StructDefinition>>,
    pub enums: Vec<AstNode<ast::EnumDefinition>>,
}

/// Parse a series of type definitions for the FFI
pub fn parse_ffi_type_defs(data: &str) -> Result<FfiTypes, ParseError> {
    let def = PolicyParser::parse(Rule::ffi_type_defs, data)?;
```